scipy/spatial/_voronoi.pyx

```diff
@@ -26,9 +26,10 @@ def sort_vertices_of_regions(int[:,::1] simplices, regions):
     cdef np.npy_intp current_simplex, current_vertex
     cdef np.npy_intp remaining_size
     cdef np.npy_intp[:] remaining
-    cdef np.npy_intp[:] sorted_vertices = np.empty(max([len(region) for region
-                                                   in regions]),
-                                                   dtype=np.intp)
+    cdef np.ndarray[np.intp_t, ndim=1] sorted_vertices
+    sorted_vertices = np.empty(max([len(region) for region
```

Comment:
you can make a for loop here with 

```python
max_len = 0
for k in regions:
    max_len = max(max_len, len(k))
```

so that you don't create a temporary list in case region numbers are high. If regions have a particular type it would be faster.

Reply:
I went ahead and applied that change. The benchmarks were neutral on it, but I also didn't have a reason not to make it if it helps for larger cases than are covered by the suite.